src/core/dbus.c

```diff
@@ -1004,8 +1008,19 @@ static void destroy_bus(Manager *m, sd_bus **bus) {
         }
 
         /* Get rid of tracked clients on this bus */
-        if (m->subscribed && sd_bus_track_get_bus(m->subscribed) == *bus)
+        if (m->subscribed && sd_bus_track_get_bus(m->subscribed) == *bus) {
+                _cleanup_strv_free_ char **subscribed = NULL;
+                int r;
+
+                r = bus_track_to_strv(m->subscribed, &subscribed);
+                if (r < 0)
+                        log_error_errno(r, "Failed to serialize api subscribers: %m");
+                r = strv_extend_strv_consume(&m->deserialized_subscribed, TAKE_PTR(subscribed), /* filter_duplicates = */ false);
```

Comment:
why append here btw? shouldn't we just replace? can it ever happen we actually merge two populated lists here?

Reply:
I'm not sure. I guess I thought this would handle either case OK.

Reply:
i would prefer if we'd not extend here, but replace. for the simple reason we should not have something that by accident can grow without boundaries. If there's no reason to extend here, then please just replace.